job/job.py

```diff
@@ -22,28 +20,24 @@ def run():
 
     try:
         EXPERIMENT_DT = datetime.now()
-
-        # Step 1: Fetch fresh data, hydrate it, and upload it to the warehouse
         fetch_and_upload_data(site, EXPERIMENT_DT)
-
-        # Step 2: Train models by pulling data from the warehouse and uploading new recommendation objects
-        top_articles = warehouse.get_default_recs(site=site)
-        save_defaults.save_defaults(top_articles, site, EXPERIMENT_DT.date())
-
         interactions_data = warehouse.get_dwell_times(site, days=config.get("DAYS_OF_DATA"))
 
-        # Temporary solution: The first position in the array of strategies is currently CFConfig.
-        # We will change this line when we implement iteration through different strategies.
-        strategy = site.strategies[0]
-        recommendations = train_model.get_recommendations(
-            interactions_data,
-            strategy.training_params,
-            EXPERIMENT_DT,
-        )
-        logging.info(f"Successfully trained model on {len(interactions_data)} inputs.")
-
-        save_predictions.save_predictions(site, recommendations)
-
+        logging.info(f"Number of strategies: {len(site.strategies)}")
+        for strategy in site.strategies:
+            logging.info(f"Strategy: {strategy.model_type}")
```

Comment:
⛏ These logging-strategy pairs could use a line break in between.

Reply:
Do you mean like this?
`logging.info(f"Number of strategies:\n{len(site.strategies)}")`

Reply:
I meant adding line breaks in the code itself to make it easier to read! Something like this:

```python
            logging.info(f"Strategy: {strategy.model_type}")
            strategy.prepare(site=site, experiment_time=EXPERIMENT_DT)

            logging.info("Fetching data...")
            strategy.fetch_data(interactions_data=interactions_data)

            logging.info("Preprocessing data...")
            strategy.preprocess_data()

            logging.info("Generating embeddings...")
            strategy.generate_embeddings()

            logging.info("Generating recommendations...")
            strategy.generate_recommendations()

            logging.info("Saving recommendations...")
            strategy.save_recommendations()

            logging.info("Job succeeded.")
```

Reply:
oh... that's easier. :)